### PR TITLE
Feat/router path handling migrations

### PR DESCRIPTION
**Pull request by kikito**
This is an extension of #5292 which also includes an automatic migration feature.

Recommended to review commit-by-commit.

Note that this PR uses the {`feat/router-path-handling` route / #5292 PR } as a target in order to make the different commits visible - hence the `pr/do not merge` label.

Please review & merge #5292 first, and then change this PR's target to `next`, removing `pr/do not merge` branch.

### CODE REVIEW

**Comment by kikito** [comment]
This needs a rebase on top of #5292, please don't merge before 5292 is reviewed and merged.

File: kong/db/migrations/core/007_140_to_200.lua [hunk 1]
```diff
@@ -23,7 +95,67 @@ return {
...
+          cassandra.text("path_handling"),
+          cassandra.text(preset_path_handling)
+        }))
+
+        local rows = assert(connector:query([[
+          SELECT id FROM routes;
+        ]]))
+        for i = 1, #rows do
+          assert(connector:query(
+            "UPDATE routes SET path_handling = ? WHERE id = ?",
```
Comment (hishamhm) [inline kong/db/migrations/core/007_140_to_200.lua]:
```
./kong/cmd/utils/migrations.lua:158: [Cassandra error] failed to run migration '007_140_to_200' up: ./kong/db/migrations/core/007_140_to_200.lua:150: [Invalid] Some partition key parts are missing: partition
```

File: kong/db/migrations/core/007_140_to_200.lua [hunk 2]
```diff
@@ -23,7 +95,67 @@ return {
...
+            VALUES(?, ?, ?)
+        ]], {
+          cassandra.text("007_140_to_200"),
+          cassandra.text("path_handling"),
+          cassandra.text(preset_path_handling)
+        }))
+
+        local rows = assert(connector:query([[
+          SELECT id FROM routes;
+        ]]))
```
Comment (hishamhm) [inline kong/db/migrations/core/007_140_to_200.lua]: This does *not* give you all routes in Cassandra, you must use the `iterate` method.

**Comment by kikito** [comment]
Closing in favor of #5360

**Pull request closed without merging**